recipes/aruco/3.x.x/conanfile.py

```diff
@@ -38,6 +38,12 @@ def requirements(self):
         self.requires("eigen/3.3.8")
         self.requires("zlib/1.2.11")
 
+        self.options[
+            "opencv"
+        ].shared = (
+            self.options.shared
+        )  # For some reason, this is not automagically inherited from self.options.shared
```

Comment:
does there need to be another check in build incase the user overrides? This is still very vague in my head, rather safe then sorry, if any one can confirm

Reply:
@bverhagen 
Will it fail if aruco is e.g. built as a static library and opencv as a shared library?
If so, then you should add a test to `build`, throwing a `ConanInvalidConfiguration` if you detect an error.

@uilianries 
Is there a conan way to do this available or on the horizon?

Reply:
No, have to build and check.

> does there need to be another check in build incase the user overrides

No, even if it was in configure(), all user options are collected and available in configure.

Reply:
I'm curious about your error, it seems be related to link order

Reply:
This is the error:

```/usr/bin/ld: bin/test_package: hidden symbol `_Z7cvTruncRKN2cv9softfloatE' in /home/conan/w/cci_PR-3646/.conan/data/opencv/4.5.0/_/_/package/74bf96ed8c4193fbaf98035f279434691d003ce6/lib/libopencv_core.a(softfloat.cpp.o) is referenced by DSO
/usr/bin/ld: final link failed: Bad value    
collect2: error: ld returned 1 exit status    
make[2]: *** [bin/test_package] Error 1    
make[1]: *** [CMakeFiles/test_package.dir/all] Error 2```

I don't think it is related to link order, but I am prepared to be proven wrong :-)

Reply:
Hmmm https://stackoverflow.com/a/23708713/8480874 never seen "hidden linkage" something new to learn!

Reply:
Is the hiding of the symbol a big in opencv (not exporting the symbol) or is this a case of using a hidden symbol (and depending on an implementation detail)?